test/extension/test_base.py

```diff
@@ -35,6 +35,7 @@
 @pytest.mark.parametrize("value,expected", parameters)
 def test_valid_colours(value, expected):
     extension = _Extension(foreground=value)
+    extension._configure(None)
     assert extension.foreground == expected
```

Comment:
Perhaps this test should also test that `extension_defaults` are being used?

Reply:
Yes, but there's no really nice way to do this.

Ideally, you'd run it in a manager instance but extensions aren't `CommandObject` so they're not visible via IPC.

We could just do a case where colours are set via `_Extension.global_defaults` (which is what `extension_defaults` are ultimately set).

Reply:
OK yeah it looks fiddly without a huge amount of benefit. I'm pretty happy with how this is now.